llvm/test/CodeGen/SPIRV/inline/type.ll

```diff
@@ -0,0 +1,43 @@
+; RUN: llc -verify-machineinstrs -O0 -mtriple=spirv-unknown-unknown %s -o - | FileCheck %s
+; RUN: %if spirv-tools %{ llc -O0 -mtriple=spirv-unknown-unknown %s -o - -filetype=obj | spirv-val %}
+; RUN: %if spirv-tools %{ llc -O0 -mtriple=spirv-unknown-unknown %s -o - | spirv-as - -o - | spirv-val %}
+
+; CHECK: [[uint32_t:%[0-9]+]] = OpTypeInt 32 0
+
+; CHECK: [[image_t:%[0-9]+]] = OpTypeImage %3 2D 2 0 0 1 Unknown
+%type_2d_image = type target("spirv.Image", float, 1, 2, 0, 0, 1, 0)
+
+%literal_true = type target("spirv.Literal", 1)
+%literal_32 = type target("spirv.Literal", 32)
+
+; CHECK: [[uint32_4:%[0-9]+]] = OpConstant [[uint32_t]] 4
+%integral_constant_4 = type target("spirv.IntegralConstant", i32, 4)
+
+; CHECK: OpUnknown(28, 4) [[array_t:%[0-9]+]] [[image_t]] [[uint32_4]]
+%ArrayTex2D = type target("spirv.Type", %type_2d_image, %integral_constant_4, 28, 0, 0)
+
+; CHECK: [[getTexArray_t:%[0-9]+]] = OpTypeFunction [[array_t]]
+
+; CHECK: OpUnknown(21, 4) [[int_t:%[0-9]+]] 32 1
+%int_t = type target("spirv.Type", %literal_32, %literal_true, 21, 0, 0)
```

Comment:
Do you think it would be useful to add a more complexe type, like CooperativeMatrix?

```llvm
%scope = type target("spirv.IntegralConstant", i32, 2) ; Workgroup
%cols = type target("spirv.IntegralConstant", i32, 4)
%rows = type target("spirv.IntegralConstant", i32, 4)
%use = type target("spirv.IntegralConstant", i32, 0) ; MatrixAKHR

%coop = type target("spirv.Type", float, %scope, %rows, %cols, %use, 4456, 0, 0)
```

Reply:
Yes, that seems like a good idea. It requires the `CooperativeMatrixKHR` capability, which I don't know how to add, so I've just added in in a new test file which doesn't run `spirv-val` for now.